### ISSUE TITLE
docs.rs build failing for main crate

### ISSUE THREAD

**Issue by pkgw** [tectonic-typesetting/tectonic#788]
We're failing to build the main crate on docs.rs; [example log](https://docs.rs/crate/tectonic/0.6.1/builds/400102).

It looks like there is an issue with [cbindgen](https://github.com/eqrion/cbindgen/) that is requiring us to hit the network during the build process, and docs.rs builds without network access.

I've filed

https://github.com/eqrion/cbindgen/issues/691

with my initial diagnosis. Depending on what's possible in cbindgen, we may have to figure out a way to work around the issue locally. There is a `DOCS_RS` environment variable that we can check in build.rs if we need to get gnarly.

**Comment by pkgw** [comment]
Also, for `tectonic_bridge_flate` and presumably other crates, we have:

```
thread 'main' panicked at 'unexpected error clearing local Cargo.lock: Read-only file system (os error 30)', 
```

This we can/should fix by updating our cbindgen build.rs boilerplate.

**Comment by pkgw** [comment]
There is a `CARGO_NET_OFFLINE` environment variable that we could try setting independently of cbindgen, but I think that using it might just cause `cargo metadata` to fail in a different way in the docs.rs build.

**Comment by pkgw** [comment]
Well, my initial testing suggests that `CARGO_NET_OFFLINE` does the trick. I'll try rolling that out under the assumption that any cbindgen fix will take a little while to emerge.

**Comment by pkgw** [comment]
Not there yet. Now:

```
error: failed to write /opt/rustwide/cargo-home/registry/src/github.com-1ecc6299db9ec823/tectonic_bridge_flate-0.1.4/Cargo.lock

Caused by:
  failed to open: /opt/rustwide/cargo-home/registry/src/github.com-1ecc6299db9ec823/tectonic_bridge_flate-0.1.4/Cargo.lock

Caused by:
  Read-only file system (os error 30)
```

Indeed, `cargo-home` is mounted read-only in the docs.rs framework.

**Comment by pkgw** [comment]
OK, finally fixed this with #793. I don't see a way to get build-time cbindgen plus workspace library crates plus docs.rs to all work together without some significant infrastructure work.
